src/web/core/src/main/java/org/geoserver/web/data/workspace/WorkspaceEditPage.java

```diff
@@ -171,10 +182,16 @@ private void saveWorkspace() {
 
         NamespaceInfo namespaceInfo = (NamespaceInfo) nsModel.getObject();
         WorkspaceInfo workspaceInfo = (WorkspaceInfo) wsModel.getObject();
-        
+
+        namespaceInfo.setIsolated(workspaceInfo.isIsolated());
+
         // sync up workspace name with namespace prefix, temp measure until the two become separate
         namespaceInfo.setPrefix(workspaceInfo.getName());
-        
+
+        // validate workspace and namespace before updating them
+        catalog.validate(workspaceInfo, false).throwIfInvalid();
```

Comment:
Better to turn this into a validator and generate a "nice" error message instead of making the page fail fully and display a stack trace?

Reply:
This is already done, the exception message will be catch and show to the user:
![image](https://user-images.githubusercontent.com/1590542/36213027-7e42ed0e-119d-11e8-8fa3-0b00339f70f6.png)